src/globus_sdk/utils.py

```diff
@@ -46,7 +46,11 @@ def safe_stringify(value: ToStr) -> str:
     if isinstance(value, str):
         return value
     if isinstance(value, bytes):
-        return value.decode("utf-8")
+        try:
+            return value.decode("utf-8")
+        except UnicodeDecodeError:
+            # this raises its own ValueError if value doesn't decode into a UUID
+            return str(uuid.UUID(bytes=value))
```

Comment:
I'd rather not make this conversion. I don't know if UUID bytes are guaranteed to _not_ be valid UTF8.

Also, it just doesn't seem very necessary to me? If a caller has the bytes for a UUID, they can do this themselves.

Reply:
Yeah, I wanted to put this in as a demonstration of "get as close to we can to stringifying whatever the user puts in". I'm similarly happy with a more stringent version.